### PR TITLE
[PORT] Mempool packages

### PR DESCRIPTION
**Pull request by ptschip**
Add tracking of ancestor packages and expose ancestor/descendant information over RPC.

### CODE REVIEW

**Comment by ptschip** [comment]
taking out of WIP

**Comment by ptschip** [comment]
Just had one more thing to add.  When you do a verbose "getrawmempool true", or getmempoolentry,  you get to see your parent and child tx's like the following, so it saves a lot of time looking them up:

 "7374fb42ba18d01d44cb0524bdf5b936d5784a0b655decb1c2d24c01f9217eeb": {
    "size": 226,
    "fee": 0.00000249,
    "modifiedfee": 0.00000249,
    "time": 1537308869,
    "height": 548450,
    "startingpriority": 0,
    "currentpriority": 0,
    "descendantcount": 2,
    "descendantsize": 226,
    "descendantfees": 249,
    "ancestorcount": 2,
    "ancestorsize": 226,
    "ancestorfees": 249,
    "depends": [
         f7c731be208d30149ca615086c4bdab94f5f9dfd31218e2deb34b839934918d3
    ],
    "spentby": [
         5de248fd3fd9753efdf4b659080c49f8b0e9829ab928c1644a60e565e9ff44e0
    ]

**Review by sickpig — APPROVED** [review]

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -292,6 +321,177 @@ UniValue getrawmempool(const UniValue &params, bool fHelp)
...
+            HelpExampleRpc("getmempooldescendants", "\"mytxid\""));
+    }
+
+    bool fVerbose = false;
+    if (params.size() > 1)
+        fVerbose = params[1].get_bool();
+
+    uint256 paramhash = ParseHashV(params[0], "parameter 1");
+
+    WRITELOCK(mempool.cs);
```
Comment (gandrewstone) [inline src/rpc/blockchain.cpp:425]: why a write lock?

**Comment by ptschip** [comment]
because _CalculateDescendants() needs a writelock, at least in the code
is says it has AssertWriteLockHelds(mempool.cs)

**Pull request merged**